dashboard/modules/reporter/reporter_agent.py

```diff
@@ -394,30 +397,64 @@ def _get_cpu_percent(in_k8s: bool):
     @staticmethod
     def _get_gpu_usage():
         global enable_gpu_usage_check
-        if gpustat is None or not enable_gpu_usage_check:
+        if pynvml is None or not enable_gpu_usage_check:
             return []
         gpu_utilizations = []
-        gpus = []
         try:
-            gpus = gpustat.new_query().gpus
+            pynvml.nvmlInit()
+            num_gpus = pynvml.nvmlDeviceGetCount()
+            for i in range(num_gpus):
+                gpu_handle = pynvml.nvmlDeviceGetHandleByIndex(i)
+                utilization = None
+                try:
+                    utilization_info = pynvml.nvmlDeviceGetUtilizationRates(gpu_handle)
+                    utilization = int(utilization_info.gpu)
+                except pynvml.NVMLError as e:
+                    logger.debug(f"pynvml failed to retrieve GPU utilization: {e}")
+
+                # processes pids
+                processes_pids = None
+                try:
+                    nv_comp_processes = pynvml.nvmlDeviceGetComputeRunningProcesses(
+                        gpu_handle
+                    )
+                    nv_graphics_processes = (
+                        pynvml.nvmlDeviceGetGraphicsRunningProcesses(gpu_handle)
+                    )
+                    processes_pids = [
+                        ProcessGPUInfo(
+                            pid=nv_process.pid,
+                            gpu_memory_usage=nv_process.usedGpuMemory // MiB
+                            if nv_process.usedGpuMemory
+                            else 0,
+                        )
+                        for nv_process in (nv_comp_processes + nv_graphics_processes)
+                    ]
```

Comment:
This is how gpustat is implemented?

Reply:
yes, calling the same pynvml methods from here
https://github.com/wookayin/gpustat/blob/64d22a9e20f6c76a85a858c87f6d7313300a6d65/gpustat/core.py#L454